### ISSUE TITLE
Vagrant 2.1.4 breaks plugin detection / install script

### ISSUE THREAD

**Issue by JohnMaguire** [hashicorp/vagrant#10178]
### Vagrant version
Vagrant 2.1.2 (downgraded) / Vagrant 2.1.4

### Host operating system
Linux 4.18.5-arch1-1-ARCH

### Guest operating system
N/A

### Vagrantfile
```ruby
# Handle installation of external plugins required for this Vagrantfile to work
required_plugins = %w(
  vagrant-s3auth
  vagrant-sshfs
)
needs_restart = false
toplevel_dir = File.dirname(File.expand_path(__FILE__))
required_plugins.each do |plugin|
  # Install any required plugins that are missing
  unless Vagrant.has_plugin? plugin
    # Plugin is not installed yet, so attempt to install it now.  Bail out on
    # failure so we don't get stuck in an infinite loop.
    # :chdir added to avoid recursively loading this file while loading plugins (https://github.com/hashicorp/vagrant/issues/9972)
    system("vagrant plugin install #{plugin} --plugin-clean-sources --plugin-source file://#{toplevel_dir}/vagrant/gems", :chdir => "/tmp") || exit!
    needs_restart = true
  end
end
if needs_restart
  # Relaunch Vagrant so the plugin is detected. Exit with the same status code.
  exec "vagrant #{ARGV.join' '}"
end
```

### Debug output
https://gist.github.com/JohnMaguire/a0ad03c37dcb57e09fc37d9689d47ce8 (Hit C-c after a few seconds)

### Expected behavior
VM brought up (as Vagrant 2.1.2 does)

### Actual behavior
Plugin install loop:

```
Installing the 'vagrant-s3auth' plugin. This can take a few minutes...
Installed the plugin 'vagrant-s3auth (1.3.2)'!
Installing the 'vagrant-sshfs' plugin. This can take a few minutes...
Installed the plugin 'vagrant-sshfs (1.3.0)'!
Installing the 'vagrant-s3auth' plugin. This can take a few minutes...
Installed the plugin 'vagrant-s3auth (1.3.2)'!
Installing the 'vagrant-sshfs' plugin. This can take a few minutes...
Installed the plugin 'vagrant-sshfs (1.3.0)'!
Installing the 'vagrant-s3auth' plugin. This can take a few minutes...
```

**Comment by pturley0** [comment]
I'm seeing the same symptoms as the original report.

Vagrant version 2.1.4
macOS High Sierra 10.13.6

**Comment by zadias** [comment]
Same here
2.1.4 on Windows 10

**Comment by r00t8** [comment]
Vagrant version 2.1.4
MacOS High Sierra 10.13.6

```
vagrant up
Installing plugins: vagrant-vbguest vagrant-hostmanager
Installing the 'vagrant-vbguest' plugin. This can take a few minutes...
Installed the plugin 'vagrant-vbguest (0.16.0)'!
Installing the 'vagrant-hostmanager' plugin. This can take a few minutes...
Installed the plugin 'vagrant-hostmanager (1.8.9)'!
Installing plugins: vagrant-vbguest vagrant-hostmanager
Installing the 'vagrant-vbguest' plugin. This can take a few minutes...
Installed the plugin 'vagrant-vbguest (0.16.0)'!
Installing the 'vagrant-hostmanager' plugin. This can take a few minutes...
Installed the plugin 'vagrant-hostmanager (1.8.9)'!
Installing plugins: vagrant-vbguest vagrant-hostmanager
Installing the 'vagrant-vbguest' plugin. This can take a few minutes...
Installed the plugin 'vagrant-vbguest (0.16.0)'!
Installing the 'vagrant-hostmanager' plugin. This can take a few minutes...
Installed the plugin 'vagrant-hostmanager (1.8.9)'!
Installing plugins: vagrant-vbguest vagrant-hostmanager
Installing the 'vagrant-vbguest' plugin. This can take a few minutes...
Installed the plugin 'vagrant-vbguest (0.16.0)'!
Installing the 'vagrant-hostmanager' plugin. This can take a few minutes...
Installed the plugin 'vagrant-hostmanager (1.8.9)'!
Installing plugins: vagrant-vbguest vagrant-hostmanager
Installing the 'vagrant-vbguest' plugin. This can take a few minutes...
Installed the plugin 'vagrant-vbguest (0.16.0)'!
Installing the 'vagrant-hostmanager' plugin. This can take a few minutes...
Installed the plugin 'vagrant-hostmanager (1.8.9)'!
Installing plugins: vagrant-vbguest vagrant-hostmanager
Installing the 'vagrant-vbguest' plugin. This can take a few minutes...
Installed the plugin 'vagrant-vbguest (0.16.0)'!
Installing the 'vagrant-hostmanager' plugin. This can take a few minutes...
Installed the plugin 'vagrant-hostmanager (1.8.9)'!
Installing plugins: vagrant-vbguest vagrant-hostmanager
Installing the 'vagrant-vbguest' plugin. This can take a few minutes...
Installed the plugin 'vagrant-vbguest (0.16.0)'!
Installing the 'vagrant-hostmanager' plugin. This can take a few minutes...
Installed the plugin 'vagrant-hostmanager (1.8.9)'!
Installing plugins: vagrant-vbguest vagrant-hostmanager
Installing the 'vagrant-vbguest' plugin. This can take a few minutes...
Installed the plugin 'vagrant-vbguest (0.16.0)'!
Installing the 'vagrant-hostmanager' plugin. This can take a few minutes...
Installed the plugin 'vagrant-hostmanager (1.8.9)'!
Installing plugins: vagrant-vbguest vagrant-hostmanager
Installing the 'vagrant-vbguest' plugin. This can take a few minutes...
```

**Comment by r00t8** [comment]
There was block of code in my VagrantFile which was forcing to install plugins, somehow it looks creating problem. I commented it to out and that fixed the issue

```
## Install plugins from the required_plugins list
#required_plugins = %w( vagrant-vbguest vagrant-hostmanager )
#plugins_to_install = required_plugins.select { |plugin| not Vagrant.has_plugin? plugin }
#if not plugins_to_install.empty?
#  puts "Installing plugins: #{plugins_to_install.join(' ')}"
#  if system "vagrant plugin install #{plugins_to_install.join(' ')}"
#    exec "vagrant #{ARGV.join(' ')}"
#  else
#    abort "Installation of one or more plugins has failed. Aborting."
#  end
#end
```

**Comment by c33s** [comment]
maybe also related to https://github.com/hashicorp/vagrant/issues/10186 
could be that because of the double loading of the vagrantfile some (environment) variables are not set, which maybe also could lead to this problem.

**Comment by Lovivan** [comment]
Same issue on 

Vagrant version 2.1.4
macOS High Sierra 10.13.6

**Comment by chrisroberts** [comment]
Fixed by #10199
